utils/update_copyright.py

```diff
@@ -17,40 +30,53 @@ def main():
     try:
         year = sys.argv[1]
         sys.stdout.write('Updating copyright year to: ' + year + ' ')
-        src_path = os.path.abspath(os.path.join(
-            os.path.dirname(__file__), '../src'))
+        base_path = os.path.abspath(
+            os.path.join(os.path.dirname(__file__), '..'))
+        src_path = os.path.join(base_path, 'src')
         # Fix copyright headers in C++ files
+        regex_header = re.compile(
+            '(.*Copyright \(C\) \d\d\d\d)(.*)( by the Widelands Development Team.*)')
+
+        def repl_header(match):
+            if match.group(1).endswith(year):
+                return match.group(1) + match.group(3)
+            else:
+                return match.group(1) + '-' + year + match.group(3)
         for filename in find_files(src_path, ['.h', '.cc']):
-            sys.stdout.write('.')
-            sys.stdout.flush()
-            lines = read_text_file(filename).strip().split('\n')
-            new_lines = []
-            regex = re.compile(
-                '(.*Copyright \(C\) \d\d\d\d)(.*)( by the Widelands Development Team.*)')
-            for line in lines:
-                match = regex.match(line)
-                if match:
-                    if match.group(1).endswith(year):
-                        line = match.group(1) + match.group(3)
-                    else:
-                        line = match.group(1) + '-' + year + match.group(3)
-                new_lines.append(line.rstrip() + '\n')
-            write_text_file(filename, ''.join(new_lines))
+            update_file(filename, regex_header, repl_header)
 
         # Now update the Buildinfo
         filename = os.path.join(src_path, 'build_info.h')
-        lines = read_text_file(filename).strip().split('\n')
-        new_lines = []
         regex = re.compile(
             '(.*constexpr uint16_t kWidelandsCopyrightEnd = )(\d\d\d\d)(;)')
-        for line in lines:
-            match = regex.match(line)
-            if match:
-                line = match.group(1) + year + match.group(3)
-            new_lines.append(line.rstrip() + '\n')
-        write_text_file(filename, ''.join(new_lines))
+
+        def replace(match): return match.group(1) + year + match.group(3)
```

Comment:
Although this is a onel-iner, please make it into two lines.

At least make an empty line after this. This would increase readability, imho.

Otherwise looks LGTM.